### ISSUE TITLE
"(deleted)" files in /data/small* directories

### ISSUE THREAD

**Issue by xor-eax-ebx** [VictoriaMetrics/VictoriaMetrics#716]
**Describe the bug**
VictoriaMetrics leaving not completely removed files 
```
lsof | grep victoria | grep -c deleted
142359
lsof | grep deleted | awk '{print $10;}'
/data/victoriametrics/data/small/2020_08/12509375_11558539_20200807065635.791_20200807065717.993_1626FBD2B11EAAC2/values.bin
/data/victoriametrics/data/small/2020_08/12509375_11558539_20200807065635.791_20200807065717.993_1626FBD2B11EAAC2/index.bin
/data/victoriametrics/data/small/2020_08/5491622_5298795_20200810064325.700_20200810064358.140_1626FBD2B128ED58/timestamps.bin
/data/victoriametrics/data/small/2020_08/5491622_5298795_20200810064325.700_20200810064358.140_1626FBD2B128ED58/values.bin
/data/victoriametrics/data/small/2020_08/377844286_30473412_20200810062718.795_20200810063744.222_1626FBD2B128E9BE/timestamps.bin
/data/victoriametrics/data/small/2020_08/377844286_30473412_20200810062718.795_20200810063744.222_1626FBD2B128E9BE/values.bin
/data/victoriametrics/data/small/2020_08/377844286_30473412_20200810062718.795_20200810063744.222_1626FBD2B128E9BE/index.bin
[...]
```
it's not a NFS - just regular setup with SSD disks
there are no errors in logs.
single-node

**To Reproduce**
- run VictoriaMetrics
- wait few days
- run lsof | grep deleted | grep victoria

**Expected behavior**
files should be removed properly

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
victoria-metrics-20200714-141946-tags-v1.38.1-0-gb442a42d8
```
single-node

**Used command-line flags**
`-httpListenAddr=:8428 -storageDataPath=/data/victoriametrics/ -memory.allowedPercent=80 -retentionPeriod=2 -search.maxUniqueTimeseries=10000000000 -search.maxConcurrentRequests=16 -loggerFormat=json`

**Comment by valyala** [comment]
@xor-eax-ebx , try upgrading VictoriaMetrics to [the latest release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases) and verify whether the issue still persist there. See [upgrade instructions](https://victoriametrics.github.io/#how-to-upgrade-victoriametrics). If the issue still persist there, then it would be great to get more information about your setup:

* The operating system used.
* Hardware configuration used (RAM size, CPU cores count).
* Filesystem used for `-storageDataPath`.

BTW, it isn't recommended setting `-search.maxUniqueTimeseries` to `10000000000` (10 billion), since this may lead to out of memory errors and high CPU usage for extended periods of time when a heavy query hits VictoriaMetrics, which selects big number of time series. The main purpose of `-search.maxUniqueTimeseries` command-line flag is to limit the maximum resource usage (RAM, CPU, disk read bandwidth) during too heavy queries. It is better to stop executing such queries as soon as possible with the error instead of trashing RAM, CPU and disk for extended periods of time.

**Comment by valyala** [comment]
It is also recommended setting up monitoring for VictoriaMetrics in order to get better visibility and debuggability for issues similar to this one - see https://victoriametrics.github.io/#monitoring .

**Comment by valyala** [comment]
@xor-eax-ebx , are there any updates on this issue?

**Comment by valyala** [comment]
It looks like the root cause of the issue has been identified and fixed in the commit 09b0f7c202f2c0e9a6b5879372fdb62d47f4a0fd .  Previously VictoriaMetrics wasn't closing files on query timeout errors. Such files were eventually deleted after [merging into bigger files](https://medium.com/@valyala/how-victoriametrics-makes-instant-snapshots-for-multi-terabyte-time-series-data-e1f3fb0e0282), but they remained opened until VictoriaMetrics restart.

The commit will be included in the next release of VictoriaMetrics.

**Comment by valyala** [comment]
FYI, the bugfix has been included in release [v1.41.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.

@xor-eax-ebx , could you upgrade VictoriaMetrics to [v1.41.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases) and verify whether the issue is gone in your case? If it still exists, then please re-open the issue.
